src/editor/actions/actions.js

```diff
@@ -245,58 +245,69 @@ export function moveCursorToNextChunk() {
 }
 
 export function createNewNotebookOnServer(options = { forkedFrom: undefined }) {
-  return (dispatch, getState) => {
+  return async (dispatch, getState) => {
     const state = getState();
-    return createNotebookRequest(state.title, state.jsmd, options)
-      .then(notebook => {
-        const message = "Notebook saved to server";
-        dispatch(
-          updateAppMessages({
-            message,
-            messageType: `NOTEBOOK_SAVED`
-          })
-        );
-        dispatch({ type: "ADD_NOTEBOOK_ID", id: notebook.id });
-        window.history.replaceState({}, "", `/notebooks/${notebook.id}`);
-        dispatch({ type: "NOTEBOOK_SAVED" });
-      })
-      .catch(() => {
-        dispatch(
-          updateAppMessages({
-            message: "Error saving notebook.",
-            messageType: "ERROR_SAVING_NOTEBOOK"
-          })
-        );
-      });
+    try {
+      const notebook = await createNotebookRequest(
+        state.title,
+        state.jsmd,
+        options
+      );
+      const message = "Notebook saved to server";
+      dispatch(
+        updateAppMessages({
+          message,
+          messageType: `NOTEBOOK_SAVED`
+        })
+      );
+      dispatch({ type: "ADD_NOTEBOOK_ID", id: notebook.id });
+      window.history.replaceState({}, "", `/notebooks/${notebook.id}`);
+      dispatch({ type: "NOTEBOOK_SAVED" });
+    } catch (err) {
```

Comment:
this try/catch really only applies to `createNotebookRequest` i think -- if that fails, we should catch, update the message, and return from the function. the rest of those dispatches and stuff can happen outside of the try/catch

Reply:
See my comment above on this.